packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -375,6 +378,85 @@ export class Toolkit extends CloudAssemblySourceBuilder {
     return templateDiffs;
   }
 
+  /**
+   * Drift Action
+   */
+  public async drift(cx: ICloudAssemblySource, options: DriftOptions): Promise<DriftCommandResult> {
+    const ioHelper = asIoHelper(this.ioHost, 'drift');
+    const selectStacks = options.stacks ?? ALL_STACKS;
+    await using assembly = await assemblyFromSource(ioHelper, cx);
+    const stacks = await assembly.selectStacksV2(selectStacks);
+
+    let drifts = 0;
+    let uncheckedResources = 0;
+    const output = [];
+
+    for (const stack of stacks.stackArtifacts) {
+      const cfn = (await (await this.sdkProvider('drift')).forEnvironment(stack.environment, Mode.ForReading)).sdk.cloudFormation();
```

Comment:
Can this be outside the loop?

Reply:
maybe just add a comment pointing out that `stack.environment` might be different